### PR TITLE
properly compute tx fee for EIP-1559 transactions

### PR DESCRIPTION
**Pull request by roberto-bayardo**

### CODE REVIEW

**Comment by roberto-bayardo** [comment]
FYI This change does not include the accounting for fee burning, but that will come next.

File: polygon/client.go [hunk 1]
```diff
@@ -349,9 +349,20 @@ func (ec *Client) getBlock(
 	loadedTxs := make([]*loadedTransaction, len(body.Transactions))
 	for i, tx := range body.Transactions {
 		txs[i] = tx.tx
+		var gasPrice *big.Int
+		if txs[i].Type() == 2 {
```
Comment (lorines) [inline polygon/client.go:353]: nit: maybe we could create a constant for `2`. 
eg. `eip1559TransactionType = 2`

Comment (roberto-bayardo) [reply polygon/client.go]: done

File: polygon/client.go [hunk 2]
```diff
@@ -933,7 +948,16 @@ type loadedTransaction struct {
...
 
-func feeOps(tx *loadedTransaction) []*RosettaTypes.Operation {
+func feeOps(tx *loadedTransaction, baseFee *big.Int) []*RosettaTypes.Operation {
+	minerEarnedAmount := new(big.Int).Set(tx.FeeAmount)
+	if tx.Transaction.Type() == eip1559TxType {
+		if baseFee == nil {
+			log.Fatalf("base fee cannot be nil for EIP-1559 transaction type")
+		}
+		// For EIP-1559 transactions, baseFee is burned and hence should not be paid to the miner.
+		minerEarnedAmount.Sub(tx.FeeAmount, baseFee)
```
Comment (yuga-cb) [inline polygon/client.go:958]: I don't think this is specific to EIP-1559? At least on L1, all of the baseFee is burned regardless of whether it is a legacy transaction or EIP-1559 transaction.

Comment (roberto-bayardo) [reply polygon/client.go:955]: yes you are correct! Updated to always burn the base fee regardless of tx type.

**Review by andrewkmin — APPROVED** [review]

**Pull request merged**